shadowsheep1/io-app
# [5] Show the user profile deletion request status on the new profile page

## Short description
This PR shows the user profile deletion request status on the new profile page made on PR #2.

NOTE: this PR is dependent of #4 .

## List of changes proposed in this pull request
The user profile page now renders a `RemoteSwitch` to show the status of an eventually pending user data deletion request, according to this #5.

| Old Profile Page  | New Profile Page of this PR |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/16268789/201059556-c81d4b73-d451-497a-9637-0a9d405b6f87.png" width="150" />  |  <img src="https://user-images.githubusercontent.com/16268789/202425825-0a78f188-f3ae-4444-b251-84f6c489f8dd.png" width="150" />  |

During the profile page component mount, the user data processing API call is done, asking for the status of an eventually user data deletion request:

```mermaid
flowchart TD;
    step1(["Redux Action `loadUserDataProcessing.request`"]) --> step2(["Saga `loadUserDataProcessingSaga`"]) --> step3(["API call `/api/v1/user-data-processing/DELETE`"])
```

The page will show this status with the following UI states:

| API call in progress  | API call fails | API call succeeds |
| ------------- | ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/16268789/202430418-c391b68d-0713-4480-ad4c-7fcdf98e589f.png" width="200" /> | <img src="https://user-images.githubusercontent.com/16268789/202430416-2eb7a939-28cf-4c38-98e7-1c3490af8d21.png" width="200" /> | <img src="https://user-images.githubusercontent.com/16268789/202430420-4ee0f671-9869-4b03-bb03-389810ebfb9a.png" width="200" /> |

If the API call fails for some reason, the `RemoteSwitch` gives the user the ability to retry it by tapping on the **retry** icon (the round blue arrow).

During a **failed** API call, the user is notified by with a `Toast`:

<img src="https://user-images.githubusercontent.com/16268789/202431228-277b1574-40b6-471e-9e51-4e3630d987f9.png" width="150" />

## How to test
Run the app and navigate to the user profile page. 
- For a user with a pending request the switch should be `ON`
- For a user without a pending request the switch should be `OFF`
- During the progress of the API call the switch should show a spinner and no tap on it could be done.
- During a failed API call, the switch should show a blue round arrow. By tapping this icon the user is able to retry the API call. The user should also see a Toast message notifying them that an error occurred.

There is also a new test case on `./ts/features/newProfile/__test__/ProfileMainScreen.test.tsx` to test this new component.
